db/fault_injection_test.cc

```diff
@@ -546,78 +546,6 @@ TEST_P(FaultInjectionTest, WriteBatchWalTerminationTest) {
   ASSERT_EQ(db_->Get(ro, "boys", &val), Status::NotFound());
 }
 
-TEST_P(FaultInjectionTest, NoDuplicateTrailingEntries) {
```

Comment:
This test is added in https://github.com/facebook/rocksdb/commit/08721293ea92394a645492dbd61d207bca5a6846. It's to verify a bug caused by (1) write fault injection returning injected error AFTER succcesful append with (2) something else.

I think a better way to inject write error is to NOT append data if injecting error so I changed the injection behavior. See fault_injection_fs.cc below. 

As a side consequence, the bug won't be triggered anymore and this test is not needed. This test will fail with my change.